src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryDetails.cs

```diff
@@ -0,0 +1,200 @@
+        /// </remarks>
+        /// <seealso cref="StructureDefinitionSummaryDetails"/>
+        /// <seealso cref="ValueSetSummaryDetails"/>
+        /// <seealso cref="ConceptMapSummaryDetails"/>
+        internal static bool Extract(IElementNavigator nav, ArtifactSummaryDetailsCollection details)
+        {
+            // if (details.IsConformanceResource())
+            if (ModelInfo.IsConformanceResource(details.GetResourceTypeName()))
+            {
+                nav.TryExtractValue(details, ConformanceCanonicalUrlKey, "url");
+                nav.TryExtractValue(details, ConformanceNameKey, "name");
+                nav.TryExtractValue(details, ConformanceStatusKey, "status");
+                return true;
+            }
+            return false;
+        }
+    }
+
+    /// <summary>For extracting summary details from a StructureDefinition resource.</summary>
+    public static class StructureDefinitionSummaryDetails
+    {
+        static readonly string StructureDefinitionTypeName = ResourceType.StructureDefinition.GetLiteral();
+
+        public static readonly string StructureDefinitionKindKey = "StructureDefinition.Kind";
+        public static readonly string StructureDefinitionConstrainedTypeKey = "StructureDefinition.ConstrainedType";
+        public static readonly string StructureDefinitionContextTypeKey = "StructureDefinition.ContextType";
+        public static readonly string StructureDefinitionBaseKey = "StructureDefinition.Base";
+
+        public static string GetStructureDefinitionKind(this IArtifactSummaryDetailsProvider details) => details[StructureDefinitionKindKey] as string;
+        public static string GetStructureDefinitionConstrainedType(this IArtifactSummaryDetailsProvider details) => details[StructureDefinitionConstrainedTypeKey] as string;
+        public static string GetStructureDefinitionContextType(this IArtifactSummaryDetailsProvider details) => details[StructureDefinitionContextTypeKey] as string;
+        public static string GetStructureDefinitionBase(this IArtifactSummaryDetailsProvider details) => details[StructureDefinitionBaseKey] as string;
+
+        /// <summary>Extract summary details from a StructureDefinition resource.</summary>
+        /// <returns><c>true</c> if the current target is a StructureDefinition, or <c>false</c> otherwise.</returns>
+        /// <remarks>The <see cref="ArtifactSummaryGenerator"/> calls this method from a <see cref="ArtifactSummaryDetailsExtractor"/> delegate.</remarks>
+        internal static bool Extract(IElementNavigator nav, ArtifactSummaryDetailsCollection details)
+        {
+            // if (details.IsStructureDefinition())
+            if (details.GetResourceTypeName() == StructureDefinitionTypeName)
+            {
+                // Explicit extractor chaining
+                if (ConformanceSummaryDetails.Extract(nav, details))
+                {
+                    nav.TryExtractValue(details, StructureDefinitionKindKey, "kind");
```

Comment:
The boolean results from TryExtractValue() are not processed into the return value of the extraction?

Reply:
Correct, usually the caller can (and will) safely ignore the value. In some circumstances, it may be useful for callers to detect if a match was found.